src/main/java/org/springframework/data/couchbase/core/mapping/CouchbaseMappingContext.java

```diff
@@ -131,12 +132,21 @@ public void setAutoIndexCreation(boolean autoCreateIndexes) {
 	/**
 	 * override method from AbstractMappingContext as that method will not publishEvent() if it finds the entity has
 	 * already been cached
-	 * 
+	 *
 	 * @param typeInformation - entity type
 	 */
 	@Override
 	protected Optional<BasicCouchbasePersistentEntity<?>> addPersistentEntity(TypeInformation<?> typeInformation) {
-		Optional<BasicCouchbasePersistentEntity<?>> entity = super.addPersistentEntity(typeInformation);
+		Optional<BasicCouchbasePersistentEntity<?>> entity = null;
+		try {
+			entity = super.addPersistentEntity(typeInformation);
+		} catch (Exception ioe) {
```

Comment:
That shouldn't be here.

Reply:
Ok. I've changed the test case that is expecting MappingException to also expect IllegalObjectAccessException.